### ISSUE TITLE
datalad install does not respect --branch option

### ISSUE THREAD

**Issue by jennydaman** [datalad/datalad#7461]
### What is the problem?

`datalad install` seems to always checkout the default branch, ignoring the value of `--branch`

### What steps will reproduce the problem?

```shell
datalad create hello-world
cd hello-world
datalad install -d . --branch on-a-branch https://github.com/jennydaman/example-datalad.git
cd example-datalad
git branch
```

Expected: `on-a-branch`
Actual: `master`

### DataLad information

```
datalad --version
datalad 0.19.2
```

```
datalad wtf
# WTF
## configuration <SENSITIVE, report disabled by configuration>
## credentials
  - keyring:
    - active_backends:
      - SecretService Keyring
      - PlaintextKeyring with no encyption v.1.0 at /home/jenni/.local/share/python_keyring/keyring_pass.cfg
    - config_file: /home/jenni/.config/python_keyring/keyringrc.cfg
    - data_root: /home/jenni/.local/share/python_keyring
## datalad
  - version: 0.19.2
## dataset
  - branches:
    - git-annex@422291a
    - master@6488e12
  - id: d420a1f0-a32d-4243-b867-c110de28d407
  - path: /home/jenni/msGithub/another-example-datalad
  - repo: AnnexRepo
## dependencies
  - annexremote: 1.6.0
  - boto: 2.49.0
  - cmd:7z: 16.02
  - cmd:annex: 10.20230626-g801c4b7
  - cmd:bundled-git: 2.41.0
  - cmd:git: 2.41.0
  - cmd:ssh: 9.3p2
  - cmd:system-git: 2.41.0
  - cmd:system-ssh: 9.3p2
  - humanize: 4.6.0
  - iso8601: 1.1.0
  - keyring: 24.2.0
  - keyrings.alt: 4.2.0
  - msgpack: 1.0.5
  - platformdirs: 3.9.1
  - requests: 2.31.0
## environment
  - LANG: en_US.UTF-8
  - LC_MESSAGES:
  - PATH: /home/jenni/opt/bin:/home/jenni/opt/CIVET/quarantines/Linux-x86_64/CIVET-2.1.0/progs:/home/jenni/opt/CIVET/quarantines/Linux-x86_64/bin:/home/jenni/opt/itksnap-4.0.0-20230220-Linux-gcc64/bin:/home/jenni/bin:/home/jenni/.local/bin:/home/jenni/.cargo/bin:/usr/lib/ccache/bin:/home/jenni/micromamba/condabin:/usr/local/sbin:/usr/local/bin:/usr/bin:/home/jenni/.local/share/gem/ruby/3.0.0/bin:/home/jenni/.yarn/bin:/home/jenni/.conda/envs/datalad/bin:/opt/cuda/bin:/opt/cuda/nsight_compute:/opt/cuda/nsight_systems/bin:/var/lib/flatpak/exports/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/usr/lib/rustup/bin
## extensions
## git-annex
  - build flags:
    - Assistant
    - Webapp
    - Pairing
    - Inotify
    - DBus
    - DesktopNotify
    - TorrentParser
    - MagicMime
    - Benchmark
    - Feeds
    - Testsuite
    - S3
    - WebDAV
  - dependency versions:
    - aws-0.22
    - bloomfilter-2.0.1.0
    - cryptonite-0.29
    - DAV-1.3.4
    - feed-1.3.2.0
    - ghc-8.10.7
    - http-client-0.7.9
    - persistent-sqlite-2.13.0.3
    - torrent-10000.1.1
    - uuid-1.3.15
    - yesod-1.6.1.2
  - key/value backends:
    - SHA256E
    - SHA256
    - SHA512E
    - SHA512
    - SHA224E
    - SHA224
    - SHA384E
    - SHA384
    - SHA3_256E
    - SHA3_256
    - SHA3_512E
    - SHA3_512
    - SHA3_224E
    - SHA3_224
    - SHA3_384E
    - SHA3_384
    - SKEIN256E
    - SKEIN256
    - SKEIN512E
    - SKEIN512
    - BLAKE2B256E
    - BLAKE2B256
    - BLAKE2B512E
    - BLAKE2B512
    - BLAKE2B160E
    - BLAKE2B160
    - BLAKE2B224E
    - BLAKE2B224
    - BLAKE2B384E
    - BLAKE2B384
    - BLAKE2BP512E
    - BLAKE2BP512
    - BLAKE2S256E
    - BLAKE2S256
    - BLAKE2S160E
    - BLAKE2S160
    - BLAKE2S224E
    - BLAKE2S224
    - BLAKE2SP256E
    - BLAKE2SP256
    - BLAKE2SP224E
    - BLAKE2SP224
    - SHA1E
    - SHA1
    - MD5E
    - MD5
    - WORM
    - URL
    - X*
  - local repository version: 10
  - operating system: linux x86_64
  - remote types:
    - git
    - gcrypt
    - p2p
    - S3
    - bup
    - directory
    - rsync
    - web
    - bittorrent
    - webdav
    - adb
    - tahoe
    - glacier
    - ddar
    - git-lfs
    - httpalso
    - borg
    - hook
    - external
  - supported repository versions:
    - 8
    - 9
    - 10
  - upgrade supported from repository versions:
    - 0
    - 1
    - 2
    - 3
    - 4
    - 5
    - 6
    - 7
    - 8
    - 9
    - 10
  - version: 10.20230626-g801c4b7
## location
  - path: /home/jenni/msGithub/another-example-datalad
  - type: dataset
## metadata.extractors
## metadata.filters
## metadata.indexers
## python
  - implementation: CPython
  - version: 3.10.12
## system
  - distribution: arch
  - encoding:
    - default: utf-8
    - filesystem: utf-8
    - locale.prefered: UTF-8
  - filesystem:
    - CWD:
      - max_pathlength: 4096
      - mount_opts: rw,relatime,compress=zstd:3,ssd,space_cache=v2,subvolid=263,subvol=/archlinux-root
      - path: /home/jenni/msGithub/another-example-datalad
      - type: btrfs
    - HOME:
      - max_pathlength: 4096
      - mount_opts: rw,relatime,compress=zstd:3,ssd,space_cache=v2,subvolid=263,subvol=/archlinux-root
      - path: /home/jenni
      - type: btrfs
    - TMP:
      - max_pathlength: 4096
      - mount_opts: rw,relatime,compress=zstd:3,ssd,space_cache=v2,subvolid=263,subvol=/archlinux-root
      - path: /tmp
      - type: btrfs
  - max_path_length: 300
  - name: Linux
  - release: 6.4.4-arch1-1
  - type: posix
  - version: #1 SMP PREEMPT_DYNAMIC Wed, 19 Jul 2023 19:19:38 +0000
```

### Have you had any success using DataLad before?

yes

**Comment by mslw** [comment]
Thanks for the report!

I can reproduce the problem. I could not debug fully, but it seems to be related to the internal machinations `install` does to its options (and by any means is a bug within `install` that needs fixing).

Personally, I use `clone` for installing subdatasets, and you can get the expected result with the following (note that `--branch` or any other git option must go at the end) :

```
datalad clone -d . https://github.com/jennydaman/example-datalad.git --branch on-a-branch
```

or, if you prefer `install`, by specifying the `-s/--source` explicitly to eliminate faulty reinterpretation:

```
datalad install -d . --branch on-a-branch -s https://github.com/jennydaman/example-datalad.git
```

**Comment by yarikoptic** [comment]
Fix is in #7463

**Comment by yarikoptic-gitmate** [comment]
Issue fixed in [`0.19.3`](https://github.com/datalad/datalad/releases/tag/0.19.3)
